pkg/controller/registry/resolver/cache.go

```diff
@@ -0,0 +1,287 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-registry/pkg/client"
+)
+
+type ClientProvider interface {
```

Comment:
This looks good, but I'm curious why we couldn't use the sourcemanager for this? The long-lived grpc connections to the catalogs are maintained there.

Reply:
Do you mean `SourceStore`? I originally wrote the interface `RegistryClientProvider` to accept that, but I added this implementation (and renamed `AsClients` to `ClientsForNamespaces`) because I wasn't sure we wanted the extra persistent connections and CatalogSource informer in the OLM operator. Happy to go either way on that, it doesn't make much difference to the cache itself.

Reply:
Maybe I’m confused about where this will live - won’t this cache be in the catalog operator to be close to the resolver?

It can wait for a follow up but I think the source store would make sense - I’m wary of having two abstractions for the same thing.

Reply:
Oh, yes, you're right. I got crossed up somehow on Friday and added this implementation unnecessarily. We can absolutely change the interface back to match the SourceStore method in a follow-up.